docs/sources/flow/reference/components/otelcol.processor.resourcedetection.md

```diff
@@ -0,0 +1,939 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/components/otelcol.processor.resourcedetection/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/otelcol.processor.resourcedetection/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/otelcol.processor.resourcedetection/
+- /docs/grafana-cloud/send-data/agent/flow/reference/components/otelcol.processor.resourcedetection/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.resourcedetection/
+labels:
+  stage: beta
+title: otelcol.processor.resourcedetection
+description: Learn about otelcol.processor.resourcedetection
+---
+
+# otelcol.processor.resourcedetection
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" version="<AGENT VERSION>" >}}
+
+`otelcol.processor.resourcedetection` detects resource information from the host
+in a format that conforms to the [OpenTelemetry resource semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions/), and appends or
+overrides the resource values in the telemetry data with this information.
+
+{{% admonition type="note" %}}
+`otelcol.processor.resourcedetection` is a wrapper over the upstream
+OpenTelemetry Collector Contrib `resourcedetection` processor. If necessary, 
+bug reports or feature requests will be redirected to the upstream repository.
+{{% /admonition %}}
+
+You can specify multiple `otelcol.processor.resourcedetection` components by giving them
+different labels.
+
+## Usage
+
+```river
+otelcol.processor.resourcedetection "LABEL" {
+  output {
+    logs    = [...]
+    metrics = [...]
+    traces  = [...]
+  }
+}
+```
+
+## Arguments
+
+`otelcol.processor.resourcedetection` supports the following arguments:
+
+Name        | Type           | Description                                                                            | Default   | Required
+----------- | -------------- | -------------------------------------------------------------------------------------- |---------- | --------
+`detectors` | `list(string)` | An ordered list of named detectors which should be ran to detect resource information. | `["env"]` | no
```

Comment:
The detectors are ran by `otelcol.processor.resourcedetection`. For example, `ecs` and `consul` are detectors. I think `should be ran` is ok because it indicates that if a detector is no listed in `detectors`, it will not be ran at all.

Reply:
"ran" is past tense, so it's probably the wrong word here. "should" implies it's optional.. you should do it.. but you don't have to.

`An ordered list of named detectors used to detect resource information.`  This leaves it open to whichever process "owns" and runs the detectors.

Reply:
Ok, thank you, I'll change it to that!